test/test_linting.jl

```diff
@@ -0,0 +1,44 @@
+using PlutoTeachingTools
+using Test
+
+using JuliaFormatter: JuliaFormatter
+using Aqua: Aqua
+using JET: JET
+using ExplicitImports: ExplicitImports
```

Comment:
I should have gone into more detail in the PR description – sorry about that!

### JuliaFormatter.jl
[JuliaFormatter.jl](https://github.com/domluna/JuliaFormatter.jl) is the most commonly used Julia code formatter. I've added a `.JuliaFormatter.toml` that specifies the [Blue code style](https://github.com/JuliaDiff/BlueStyle) in #47. Code can be formatted by calling `using JuliaFormatter; format(".")` or by used the formatting command in VSCode. JuliaFormatter can also be used as a formatting check in the package tests. This way, we ensure that every PR commits formatted code.

### Aqua.jl
[Aqua.jl](https://github.com/JuliaTesting/Aqua.jl) runs several automated tests:

> * There are no method ambiguities.
> * There are no undefined `exports`.
> * There are no unbound type parameters.
> * There are no stale dependencies listed in `Project.toml`.
> * Check that test target of the root project `Project.toml` and test project (`test/Project.toml`) are consistent.
> * Check that all external packages listed in `deps` have corresponding `compat` entries.
> * There are no "obvious" type piracies.

### JET.jl
[JET.jl](https://github.com/aviatesk/JET.jl) automatically tests code for type instabilities (and in my experience also catches a bunch of other bugs).

### ExplicitImports.jl
[ExplicitImports.jl](https://github.com/ericphanson/ExplicitImports.jl) is a rather new project and tests proper import and export "hygiene". With the introduction of the [new `public` keyword](https://www.youtube.com/watch?v=2o8MhoN-3NE), Julia developers will hopefully avoid using non-public internals.

Reply:
@adrhill If you set up these style checks, you should also report what exactly doesn't satisfy the rules. Otherwise you create a barrier to contributions. I've not set up JuliaFormatter locally and I don't intend to do so.

Reply:
by comment is addressed here: #53

Reply:
Yes, this is a valid concern. We might also want to link to the [ColPrac guide](https://github.com/SciML/ColPrac) to help newcomers.

> I've not set up JuliaFormatter locally and I don't intend to do so.

You probably do have JuliaFormatter available within VSCode:
<img width="720" alt="image" src="https://github.com/user-attachments/assets/0014b5f6-c459-4f5c-b9cc-32f5401d57a1">

If you don't use VSCode, it is part of the PlutoTeachingTools test environment, which can be activated via `]activate test`.

The advantages and disadvantages of code formatters have been discussed to death. `gofmt` has been a big success in the Go community. If all code you read and write in a language is formatted in a standardized way, it arguably lowers the barrier to contributions.

Reply:
I am all for formatted code. What I am against is marking PRs with a red x with no instructions how to fix it.

This should be fixed if #53 works as intended.